File: python/src/adapter/driving/well_known_test_request_controller.py
import logging

from fastapi import APIRouter, Request

from src.adapter.driving.result import Result

logger = logging.getLogger(__name__)
router = APIRouter()


@router.post("/.well-known/test/request/string")
async def test_request_string(request: Request):
    """测试请求，获取请求体字符串"""
    request_body = await request.body()
    request_string = request_body.decode()
    return Result.ok(data=request_string)


@router.post("/.well-known/test/request/json")
async def test_request_json(request: Request):
    """测试请求，获取请求体 json"""
    request_json = await request.json()  # 无法解析JSON时报错: "Expecting value: line 1 column 1 (char 0)"
    return Result.ok(data=request_json)


@router.post("/.well-known/test/request/get")
async def test_request_get(request: Request):
    """测试请求，获取请求体 json 中的字段"""
    request_json = await request.json()
    test = str(request_json.get("test", "")).strip()
    return Result.ok(data=test)


@router.get("/.well-known/test/request/param")
async def test_request_param(id: str):
    """测试请求，获取 id"""
    return Result.ok(data=id)

@router.get("/.well-known/test/request/param-default")
async def test_request_param_default(id: str = ""):
    """测试请求，获取 id"""
    return Result.ok(data=id)

@router.get("/.well-known/test/request/param-none")
async def test_request_param_none(id: str | None = None):
    """测试请求，获取 id"""
    return Result.ok(data=id)

File: dotnet/src/adapter/driving/WellKnownTestRequestController.cs
using System.Text;
using System.Text.Json;

using Microsoft.AspNetCore.Mvc;

namespace DddExample.Adapter.Driving
{
    /// <summary>
    /// 测试请求 Well Known Controller
    /// </summary>
    public class WellKnownTestRequestController: ControllerBase
    {
        [HttpPost("/.well-known/test/request/string")]
        async public Task<Result> TestRequestString()
        {
            using StreamReader reader = new StreamReader(Request.Body, Encoding.UTF8);
            string body = await reader.ReadToEndAsync();
            return Result.OkData(body);
        }

        [HttpPost("/.well-known/test/request/from-body-string")]
        public Result TestRequestFromBodyString([FromBody] string body)
        {
            return Result.OkData(body);
        }

        [HttpPost("/.well-known/test/request/from-body-json")]
        public Result TestRequestFromBodyJson([FromBody] JsonElement? body)
        {
            //[FromBody] JsonElement body 只处理 application/json、text/json 以及类似的 JSON 类型
            //string name = body.Value.GetProperty("name").GetString() ?? "";
            if (body == null)
                return Result.Ok();
            else
                return Result.OkData(body);
        }

        [HttpPost("/.well-known/test/request/from-body-dynamic")]
        public Result TestRequestFromBodyDynamic([FromBody] dynamic? body)
        {
            return Result.OkData(body);
        }
    }
}
